bindings/rust/wasmedge-sys/src/instance/function.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{wasmedge, Value, WasmFnIO, HOST_FUNCS};
+
+use core::ffi::c_void;
+use std::convert::TryInto;
+
+use rand::Rng;
+
+extern "C" fn wraper_fn(
+    key_ptr: *mut c_void,
+    _data: *mut c_void,
+    _mem_cxt: *mut wasmedge::WasmEdge_MemoryInstanceContext,
+    params: *const wasmedge::WasmEdge_Value,
+    param_len: u32,
+    returns: *mut wasmedge::WasmEdge_Value,
+    return_len: u32,
+) -> wasmedge::WasmEdge_Result {
+    let key = key_ptr as *const usize as usize;
+    let mut result: Result<Vec<Value>, u8> = Err(0);
+
+    let mut input: Vec<Value> = {
+        let raw_input = unsafe {
+            std::slice::from_raw_parts(
+                params,
+                param_len
+                    .try_into()
+                    .expect("len of params should not greater than usize"),
+            )
+        };
+        raw_input.iter().map(|r| (*r).into()).collect()
+    };
+    input.remove(0);
```

Comment:
What does this `0` mean?

Reply:
The `0` one is the function reference itself, and the real input parameters start from `1`.

Reply:
Should we have a comment to explain this? Because it's hard to figure out this from the source code directly.